src/base/net/stratum/SelfSelectClient.cpp

```diff
@@ -235,6 +238,35 @@ void xmrig::SelfSelectClient::submitBlockTemplate(rapidjson::Value &result)
     });
 }
 
+int64_t xmrig::SelfSelectClient::submit(const JobResult& result)
+{
+    if (m_submit_to_origin) {
+        this->submit_origin_daemon(result);
+    }
+    return m_client->submit(result);
+}
+
+void xmrig::SelfSelectClient::submit_origin_daemon(const JobResult& result)
```

Comment:
We will probably need to add in the dero and xmrig proxy project code here as well for completeness (when we submit a PR to XMRig itself). This is fine for now though.